### PR TITLE
🎉 Source Salesforce: Add Transform for output records

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*In this pull request, `Transform` for Salesforce connector has been added in order to prevent cases when we can receive incorrect data for records of the data type `anyType` and `calculated`..*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 1]
```diff
@@ -273,7 +247,7 @@ def request_params(
             selected_properties = {
                 key: value
                 for key, value in selected_properties.items()
-                if not (("format" in value and value["format"] == "base64") or ("object" in value["type"] and len(value["type"]) < 3))
+                if not (("format" in value and value["format"] == "base64") or "object" in value["type"])
```
Comment (avida) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:250]: what a bloody condition to understand, can we do like this:
```python
if value.get("format") != "base64" and "object" not in value["type"]:
```
seems like the same condition but better to read, could you check it out as well?

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 2]
```diff
@@ -103,6 +105,18 @@ class BulkSalesforceStream(SalesforceStream):
...
+    transformer = TypeTransformer(TransformConfig.CustomSchemaNormalization | TransformConfig.DefaultSchemaNormalization)
+
+    @transformer.registerCustomTransform
+    def transform_empty_string_to_none(instance, schema):
+        """
+        BULK API returns a `csv` file, where all values are initially as string type.
+        This custom transformer replaces empty lines with `None` value.
+        """
+        if isinstance(instance, str) and not instance.strip():
+            instance = None
```
Comment (avida) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:117]: just a nitpick:
```python
return None
```

Comment (sergei-solonitcyn) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:117]: I would leave it as it is now

**Comment by sergei-solonitcyn** [comment]
We have 2 checks failed. 
If it has been already fixed in **_master_**, let's do rebase (or merge from **_master_**) and recreate the PR.

**Review by sergei-solonitcyn — CHANGES_REQUESTED** [review]
We have 2 checks failed. 
If it has been already fixed in **_master_**, let's do rebase (or merge from **_master_**) and recreate the PR.

**Review by Zirochkaa — APPROVED** [review]

**Review by sergei-solonitcyn — APPROVED** [review]

**Pull request merged**